pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_class_controller.go

```diff
@@ -45,18 +47,22 @@ func (r *GatewayClassReconciler) Reconcile(ctx context.Context, req kube_ctrl.Re
 		return kube_ctrl.Result{}, err
 	}
 
-	inUse := len(gateways.Items) > 0
-
-	if controllerutil.ContainsFinalizer(class, gatewayapi.GatewayClassFinalizerGatewaysExist) == inUse {
-		return kube_ctrl.Result{}, nil
-	}
-
-	if inUse {
+	if len(gateways.Items) > 0 {
 		controllerutil.AddFinalizer(class, gatewayapi.GatewayClassFinalizerGatewaysExist)
```

Comment:
I guess AddFinalizer is idempotent which is why you removing the guard above?

Reply:
Exactly, and keeping the early return would have prevented adding the status if e.g. it was just created and no Gateways existed yet.